src/pages/Explore/LogExploration.tsx

```diff
@@ -115,33 +115,36 @@ export class LogExploration extends SceneObjectBase<LogExplorationState> {
     this.updateVariableHide(filtersVar);
 
     // Some scene elements publish this
-    this.subscribeToEvent(StartingPointSelectedEvent, this._handleStartingPointSelected.bind(this));
-    this.subscribeToEvent(DetailsSceneUpdated, this._handleDetailsSceneUpdated.bind(this));
-
-    this.subscribeToState((newState, oldState) => {
-      if (newState.showDetails !== oldState.showDetails) {
-        if (newState.showDetails) {
-          this.state.body.setState({ secondary: new DetailsScene(this.state.detailsScene?.state || {}) });
-          this.setState({ detailsScene: undefined });
-        } else {
-          this.state.body.setState({ secondary: undefined });
-          this.setState({ detailsScene: new DetailsScene({}) });
+    this._subs.add(this.subscribeToEvent(StartingPointSelectedEvent, this._handleStartingPointSelected.bind(this)));
+    this._subs.add(this.subscribeToEvent(DetailsSceneUpdated, this._handleDetailsSceneUpdated.bind(this)));
+
+    this._subs.add(
+      this.subscribeToState((newState, oldState) => {
+        if (newState.showDetails !== oldState.showDetails) {
+          if (newState.showDetails) {
+            this.state.body.setState({ secondary: new DetailsScene(this.state.detailsScene?.state || {}) });
+            this.setState({ detailsScene: undefined });
+          } else {
+            this.state.body.setState({ secondary: undefined });
+            this.setState({ detailsScene: new DetailsScene({}) });
+          }
         }
-      }
 
-      if (newState.mode !== oldState.mode) {
-        this.setState({ topScene: getTopScene(newState.mode) });
-      }
+        if (newState.mode !== oldState.mode) {
+          this.setState({ topScene: getTopScene(newState.mode) });
+        }
 
-      const patternsVariable = sceneGraph.lookupVariable(VAR_PATTERNS, this);
-      if (patternsVariable instanceof CustomVariable) {
-        const patternsLine =
-          newState.patterns?.map((p) => `${p.type === 'include' ? '|> ' : '!> '} \`${p.pattern}\``)?.join(' ') || '';
-        patternsVariable.changeValueTo(patternsLine);
-      }
-    });
+        const patternsVariable = sceneGraph.lookupVariable(VAR_PATTERNS, this);
+        if (patternsVariable instanceof CustomVariable) {
+          const patternsLine =
+            newState.patterns?.map((p) => `${p.type === 'include' ? '|> ' : '!> '} \`${p.pattern}\``)?.join(' ') || '';
+          patternsVariable.changeValueTo(patternsLine);
+        }
+      })
+    );
 
     return () => {
+      this._subs.unsubscribe();
```

Comment:
If we are returning a custom deactivation handler, do we also need to manually unsubscribe this._subs?